lib/pbench/server/database/database.py

```diff
@@ -14,10 +16,11 @@ def get_engine_uri(config, logger):
             psql = config.get("Postgres", "db_uri")
             return psql
         except (NoSectionError, NoOptionError):
+            msg = "Failed to find a [Postgres] section in configuration file."
             if logger:
-                logger.error(
-                    "Failed to find a [Postgres] section in configuration file."
-                )
+                logger.error(msg)
+            else:
+                print(msg, file=sys.stderr)
```

Comment:
I don't want to gate the jam session, but, since we seem to be investing some effort into this, it seems like, instead of printing directly to `stderr` when `logger` is absent, we should be using a default logger (which will presumably print to `stderr`).

I haven't done the full investigation, but if we just invoke the "root" logger, then I think we'll get the ["logger of last resort"](https://docs.python.org/3/library/logging.html#logging.lastResort), which will do what we want...I think.

Reply:
That is, if `logger` is `None`, I think we can just instantiate a default instance of the `logging` class and use that instead, and then in common code issue the `error()` method call.